### PR TITLE
DOC: scipy.optimize: Added an example for wrapping complex-valued functions

### PR DESCRIPTION
**Pull request by mp4096**
See #6506

### CODE REVIEW

File: scipy/optimize/_lsq/least_squares.py [hunk 1]
```diff
@@ -255,7 +255,10 @@ def least_squares(
         ``fun(x, *args, **kwargs)``, i.e., the minimization proceeds with
         respect to its first argument. The argument ``x`` passed to this
         function is an ndarray of shape (n,) (never a scalar, even for n=1).
-        It must return a 1-d array_like of shape (m,) or a scalar.
+        It must return a 1-d array_like of shape (m,) or a scalar. If the
+        argument ``x`` is complex or the function ``fun`` returns complex
+        residuals, it must be wrapped in a real function of real arguments,
+        as shown in the `Examples`_ section.
```
Comment (person142) [inline scipy/optimize/_lsq/least_squares.py]: This link doesn't work right in the CircleCI build:

https://1324-1460385-gh.circle-artifacts.com/0/home/ubuntu/scipy/doc/build/html-scipyorg/generated/scipy.optimize.least_squares.html#id22

**Comment by person142** [comment]
Other than my one small comment LGTM!

**Comment by mp4096** [comment]
Ugh, I _seriously_ underestimated rST and Sphinx. I think it would be better to omit the link altogether. As far as I can see cross-referencing within a docstring isn't used often in SciPy docstrings.

Also adjusted the wording a little bit.

**Comment by person142** [comment]
Note that you can check this stuff locally so long as you have the right version of sphinx--1.2.3 should work. You can run `make html-scipyorg` in `doc` to build the docs; I know the README says otherwise but I'm not sure if running `python setup.py build_sphinx` actually works. Can also check the refguide with `python runtests.py --refguide-check` (though there are lots of spurious failures in python 3).

**Comment by mp4096** [comment]
Sorry, was my first time dealing with docs using sphinx + doctest. `python setup.py build_sphinx` actually worked as a charm, it was the doctest check that I'd no idea about.

Tested it locally, hope Travis CI will be the same opinion.

**Comment by person142** [comment]
> `python setup.py build_sphinx` actually worked like a charm

That's good to know.

Anyway, LGTM now.

**Pull request merged**